projects/plugins/jetpack/modules/subscriptions/subscribe-modal/subscribe-modal.js

```diff
@@ -2,26 +2,24 @@ const { domReady } = wp;
 
 domReady( function () {
 	const modal = document.getElementsByClassName( 'jetpack-subscribe-modal' )[ 0 ];
+	const modalDismissedCookie = 'jetpack_post_subscribe_modal_dismissed';
+	const hasModalDismissedCookie =
+		document.cookie && document.cookie.indexOf( modalDismissedCookie ) > -1;
 
-	if ( ! modal ) {
+	if ( ! modal || hasModalDismissedCookie ) {
 		return;
 	}
 
 	const close = document.getElementsByClassName( 'jetpack-subscribe-modal__close' )[ 0 ];
-	const modalDismissedCookie = 'jetpack_post_subscribe_modal_dismissed';
-	const hasModalDismissedCookie =
-		document.cookie && document.cookie.indexOf( modalDismissedCookie ) > -1;
 	let hasLoaded = false;
 	let isScrolling;
 
 	window.onscroll = function () {
 		window.clearTimeout( isScrolling );
```

Comment:
I'm not a fan of attaching events like this. I can easily imagine some other plugins are doing it this way and we would overwrite each other. I would use `addEventListener`/`removeEventListener` as I did for the Escape button handling. WDYT?

Reply:
Good catch! Go for it 👍

Reply:
In a follow-up PR just to make it easier to review, thanks!